src/client.ts

```diff
@@ -247,7 +257,21 @@ export function createServiceProxy<T>(
               );
             }
           });
-          return program;
+
+          return program.pipe(
+            Effect.tapError((e) =>
+              Effect.logDebug("AWS Request Failed, Retrying...", {
+                error: e,
+              }),
+            ),
+            Effect.retry({
+              schedule: Schedule.intersect(
+                Schedule.exponential("100 millis"),
+                Schedule.recurs(5),
+              ),
+              while: (e) => retryableErrorTags.includes(e._tag),
+            }),
+          );
```

Comment:
Is there a way we can make this configurable or injectable as a layer with a sensible default.